src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -219,9 +227,19 @@ function ImplicitEquationJacobian(
                 similar(Y.c, TridiagonalRow) : FT(-1) * I,
         )
     else
-        MatrixFields.unrolled_map(
-            name -> (name, name) => FT(-1) * I,
-            (diffused_scalar_names..., @name(c.uₕ)),
+        (
+            MatrixFields.unrolled_map(
+                name -> (name, @name(c.ρ)) => similar(Y.c, TridiagonalRow),
+                available_tracer_names_precip,
+            )...,
```

Comment:
Instead of initializing these blocks and setting them to 0, let's just remove them altogether.